From 2c7950b7bdebbf986b18ad09e18a168bf5ddc3eb Mon Sep 17 00:00:00 2001
From: P Praneesh <ppranees@codeaurora.org>
Date: Tue, 16 Jun 2020 17:20:43 +0530
Subject: [PATCH] ath11k: war firmware pdevid for phyb2g hw mode

For phyb2g mode firmware sends pdevid 1 instead of
pdevid 2. since firmware changes are not yet merged
add this WAR for fixed pdevid 2 if hardware mode is PHYB2G.

Signed-off-by: P Praneesh <ppranees@codeaurora.org>
---
 drivers/net/wireless/ath/ath11k/wmi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath11k/wmi.c b/drivers/net/wireless/ath/ath11k/wmi.c
index c6939b8..479b5ed 100644
--- a/drivers/net/wireless/ath/ath11k/wmi.c
+++ b/drivers/net/wireless/ath/ath11k/wmi.c
@@ -359,7 +359,11 @@ ath11k_pull_mac_phy_cap_svc_ready_ext(struct ath11k_pdev_wmi *wmi_handle,
 
 	mac_phy_caps = wmi_mac_phy_caps + phy_idx;
 
-	pdev->pdev_id = mac_phy_caps->pdev_id;
+	if (hw_mode_id == WMI_HOST_HW_MODE_2G_PHYB)
+		pdev->pdev_id = WMI_HOST_PDEV_ID_2;
+	else
+		pdev->pdev_id = mac_phy_caps->pdev_id;
+
 	pdev_cap->supported_bands = mac_phy_caps->supported_bands;
 	pdev_cap->ampdu_density = mac_phy_caps->ampdu_density;
 
-- 
2.7.4

